python/seldon_core/batch_processor.py

```diff
@@ -425,19 +425,21 @@ def _send_batch_predict_multi_request(
                 break
             except (requests.exceptions.RequestException, AssertionError) as e:
                 logger.error(
-                    f"Exception: {e}, retries {i+1} / {retries} for entries {indices}"
+                    f"Exception: {e}, retries {i+1} / {retries} for batch_id(s)={indices}"
                 )
                 if i == (retries - 1):
                     raise
 
     except Exception as e:
-        error_resp = {
-            "status": {"info": "FAILURE", "reason": str(e), "status": 1},
-            "meta": tags,
-        }
-        logger.error(f"Exception: {e}")
-        str_output = json.dumps(error_resp)
-        return [str_output]
+        output = []
+        for batch_index, batch_instance_id in zip(indices, instance_ids):
+            error_resp = {
+                "status": {"info": "FAILURE", "reason": str(e), "status": 1},
+                "meta": dict(batch_index=batch_index, batch_instance_id=batch_instance_id, **tags),
+            }
+            logger.error(f"Exception: {e}")
```

Comment:
You normally can just use logger.exception, as that prints the stack trace but this should be ok as well